### PR TITLE
do_pml should not be parsed anymore.

### PR DESCRIPTION
**Pull request by RevathiJambunathan**
This PR cleans up some old code that was added to support the old and new boundary interface.
Now that we support only the new boundary.field_lo and boundary.field_hi, do_pml should not be parsed anymore.

Consistently -- all the input files that had `warpx.do_pml`have been fixed such that this is not provided in the input as well.

### CODE REVIEW

File: Source/WarpX.cpp [hunk 1]
```diff
@@ -737,39 +738,9 @@ WarpX::ReadParameters ()
...
-                if ( Geom(0).isPeriodic(idim) == 0) parse_do_pml_Hi[idim] = 1;
-            }
-        }
-        pp_warpx.queryarr("do_pml_Hi", parse_do_pml_Hi);
-        do_pml_Hi[0] = parse_do_pml_Hi[0];
-        do_pml_Hi[1] = parse_do_pml_Hi[1];
-#if (AMREX_SPACEDIM == 3)
-        do_pml_Hi[2] = parse_do_pml_Hi[2];
-#endif
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE( isAnyBoundaryPML == false,
```
Comment (RevathiJambunathan) [inline Source/WarpX.cpp]:
```suggestion
        AMREX_ALWAYS_ASSERT_WITH_MESSAGE( isAnyBoundaryPML() == false,
```

File: Source/WarpX.cpp [hunk 2]
```diff
@@ -666,14 +666,15 @@ WarpX::ReadParameters ()
             }
         }
 
-        pp_warpx.query("do_pml", do_pml);
+        if ( pp_warpx.query("do_pml", do_pml) )
```
Comment (dpgrote) [inline Source/WarpX.cpp]: This should go in the `BackwardCompatibility` routine.

Comment (RevathiJambunathan) [reply Source/WarpX.cpp]: Thank you @dpgrote ! fixed this now

File: Source/WarpX.cpp [hunk 3]
```diff
@@ -1234,7 +1203,9 @@ WarpX::BackwardCompatibility ()
         amrex::Abort("warpx.use_kspace_filter is not supported anymore. "
                      "Please use the flag use_filter, see documentation.");
     }
-
+    if ( pp_warpx.query("do_pml", backward_int) ) {
+        amrex::Abort( "do_pml is not supported. Please use boundary.field_lo and boundary.field_hi to set the boundary conditions.") ;
```
Comment (RevathiJambunathan) [inline Source/WarpX.cpp]:
```suggestion
        amrex::Abort( "do_pml is not supported. Please use boundary.field_lo and boundary.field_hi to set the boundary conditions.");
```

File: Python/pywarpx/picmi.py [hunk 4]
```diff
@@ -525,7 +525,6 @@ class ElectromagneticSolver(picmistandard.PICMI_ElectromagneticSolver):
     def init(self, kw):
         assert self.method is None or self.method in ['Yee', 'CKC', 'PSATD'], Exception("Only 'Yee', 'CKC', and 'PSATD' are supported")
 
-        self.do_pml = kw.pop('warpx_do_pml', None)
         self.pml_ncell = kw.pop('warpx_pml_ncell', None)
```
Comment (ax3l) [inline Python/pywarpx/picmi.py:528]: @dpgrote independent of this PR, should we throw a warning in PICMI `init` methods about unused `kw` keys?
I think we would silently ignore additional arguments here, which can be hard to debug for users.

I think we basically just need to check that `kw` is empty at the end of this function, no?

File: Source/WarpX.cpp [hunk 5]
```diff
@@ -1234,7 +1203,9 @@ WarpX::BackwardCompatibility ()
         amrex::Abort("warpx.use_kspace_filter is not supported anymore. "
                      "Please use the flag use_filter, see documentation.");
     }
-
+    if ( pp_warpx.query("do_pml", backward_int) ) {
+        amrex::Abort( "do_pml is not supported. Please use boundary.field_lo and boundary.field_hi to set the boundary conditions.");
```
Comment (ax3l) [inline Source/WarpX.cpp]:
```suggestion
        amrex::Abort( "do_pml is not supported anymore. Please use boundary.field_lo and boundary.field_hi to set the boundary conditions.");
```

File: Python/pywarpx/picmi.py (hunk 4 above)
Comment (dpgrote) [reply Python/pywarpx/picmi.py:528]: This is already done, in the pimcistandard code. It happens at line 47 in the file `picmi/PICMI_Python/base.py`.

Comment (ax3l) [reply Python/pywarpx/picmi.py:528]: Ah, very cool!
https://github.com/picmi-standard/picmi/blob/4bb9858dd6b3c2e80b4e06c62a11a53b3ce9a70e/PICMI_Python/base.py#L47-L48

**Review by ax3l — APPROVED** [review]
LGTM, thanks a lot!

File: Docs/source/usage/parameters.rst [hunk 6]
```diff
@@ -211,7 +211,7 @@ Domain Boundary Conditions
 --------------------------
 
 * ``boundary.field_lo`` and ``boundary.field_hi`` (`2 strings` for 2D, `3 strings` for 3D, `pml` by default)
-    Boundary conditions applied to field at the lower and upper domain boundaries. Depending on the type of boundary condition, the value for ``geometry.is_periodic`` will be set, overriding the user-input for the input parameter, ``geometry.is_periodic``. If not set, the default value for the fields at the domain boundary will be set to pml.
+    Boundary conditions applied to field at the lower and upper domain boundaries. Depending on the type of boundary condition, the value for ``geometry.is_periodic`` will be set.
```
Comment (RevathiJambunathan) [inline Docs/source/usage/parameters.rst]:
```suggestion
    Boundary conditions applied to fields at the lower and upper domain boundaries. 
```

File: Docs/source/usage/parameters.rst [hunk 7]
```diff
@@ -211,7 +211,7 @@ Domain Boundary Conditions
 --------------------------
 
 * ``boundary.field_lo`` and ``boundary.field_hi`` (`2 strings` for 2D, `3 strings` for 3D, `pml` by default)
-    Boundary conditions applied to field at the lower and upper domain boundaries. Depending on the type of boundary condition, the value for ``geometry.is_periodic`` will be set, overriding the user-input for the input parameter, ``geometry.is_periodic``. If not set, the default value for the fields at the domain boundary will be set to pml.
+    Boundary conditions applied to fields at the lower and upper domain boundaries.
```
Comment (RevathiJambunathan) [inline Docs/source/usage/parameters.rst]:
```suggestion
    Boundary conditions applied to fields at the lower and upper domain boundaries.
```

**Review by dpgrote — APPROVED** [review]
Looks good to me!

**Pull request merged**